Common/Securities/CashAccountMarginModel.cs

```diff
@@ -0,0 +1,242 @@
+                var orderBaseCurrency = orderSecurity as IBaseCurrencySymbol;
+
+                if (baseCurrency != null && orderBaseCurrency != null)
+                {
+                    var orderPrice = 0m;
+                    switch (order.Type)
+                    {
+                        case OrderType.Limit:
+                            orderPrice = ((LimitOrder) order).LimitPrice;
+                            break;
+                        case OrderType.StopMarket:
+                            orderPrice = ((StopMarketOrder) order).StopPrice;
+                            break;
+                        case OrderType.StopLimit:
+                            orderPrice = ((StopLimitOrder) order).LimitPrice;
+                            break;
+                    }
+
+                    // convert order value to target currency
+                    decimal orderValue;
+                    if (orderSecurity.QuoteCurrency.Symbol == targetCurrency)
+                    {
+                        orderValue = orderPrice * order.Quantity;
+                        if (order.Direction == OrderDirection.Buy)
+                        {
+                            orderValue = -orderValue;
+                        }
+                    }
+                    else
+                    {
+                        orderValue = order.GetValue(orderSecurity) / orderPrice;
+                    }
+
+                    openOrdersTotalValue += orderValue;
+                }
+                else
+                {
+                    openOrdersTotalValue += order.GetValue(orderSecurity);
+                    openOrdersTotalQuantity += order.Quantity;
+                }
+            }
+
+            switch (direction)
+            {
+                case OrderDirection.Hold:
+                case OrderDirection.Buy:
+                    // increasing position, purchasing in units of the quote currency
+                    return security.QuoteCurrency.Amount - Math.Abs(openOrdersTotalValue);
+
+                case OrderDirection.Sell:
+                    // remaining margin in units of the base currency ( can't sell what we don't have )
+                    if (baseCurrency != null)
+                    {
+                        // we have this much 'cash' that can be sold
+                        return _cashBook[baseCurrency.BaseCurrencySymbol].Amount - Math.Abs(openOrdersTotalValue);
+                    }
+
+                    // we have this much stock value that can be sold... this is in the account currency,
+                    // the assumption being that since the security doesn't implement IBaseCurrencySymbol
+                    // that the holdings are either 'units' of stock or similar and not currency swaps/virtual positions
+                    return (security.Holdings.AbsoluteQuantity - Math.Abs(openOrdersTotalQuantity)) * security.Holdings.AveragePrice;
```

Comment:
This math looks wrong. We have a quantity and subtract from it cost. This can return units of quantity or value. I think the more correct thing is to return quantity. You can sell 5 units of SPY, rather than you can sell 1,000 USD of SPY. The important part is that the other methods in this implementation are consistent.